Parsers/ASimAuditEvent/Parsers/vimAuditEventSentinelOne.yaml

```diff
@@ -0,0 +1,713 @@
+                            "Set"
+                        );
+        let parsedotheractivitydata_objectvalue = rawotheractivitydata
+            | where activityType_d in (3008, 3009, 3010, 3011, 3012, 3013, 3014, 3015, 3016, 3017, 3018, 3019, 3650, 3651, 3652, 3653, 3654)
+            | extend Object = strcat(Object, ' ', value);
+        let parsedotheractivitydata_severity = rawotheractivitydata
+            | where activityType_d in (2036, 2037, 2030)
+            | where (eventresult == "*" or EventResult =~ eventresult)
+                and (array_length(eventtype_in) == 0 or EventType has_any (eventtype_in))
+                and (array_length(operation_has_any) == 0 or Operation has_any (operation_has_any))
+                and (array_length(newvalue_has_any) == 0 or NewValue has_any (newvalue_has_any))
+                and (array_length(object_has_any) == 0 or Object has_any (object_has_any))
+                and (array_length(srcipaddr_has_any_prefix) == 0 or has_any_ipv4_prefix(ipAddress, srcipaddr_has_any_prefix))
+            | extend EventSeverity_specific = case(
+                                        primaryDescription_s has_any ("to malicious", "to True positive"),
+                                        "High", 
+                                        primaryDescription_s has_any ("to suspicious", "to Undefined"),
+                                        "Medium",
+                                        primaryDescription_s has "to False positive",
+                                        "Low",
+                                        "Informational"
+                                    );
+        let ParsedActivitydata = union
+                groupsiteactivitydata,
+                machineactivitydata,
+                accountactivitydata,
+                useractivitydata,
+                rawotheractivitydata,
+                parsedotheractivitydata_eventtype,
+                parsedotheractivitydata_objectvalue
+            | where activityType_d !in(2030, 2036, 2037)
+            | lookup EventSeverityLookup on EventResult
+            | lookup EventSeverityLookup_activity on activityType_d
+            | where (eventresult == "*" or EventResult =~ eventresult)
+                and (array_length(eventtype_in) == 0 or EventType has_any (eventtype_in))
+                and (array_length(operation_has_any) == 0 or Operation has_any (operation_has_any))
+                and (array_length(newvalue_has_any) == 0 or NewValue has_any (newvalue_has_any))
+                and (array_length(object_has_any) == 0 or Object has_any (object_has_any))
+                and (array_length(srcipaddr_has_any_prefix) == 0 or has_any_ipv4_prefix(ipAddress, srcipaddr_has_any_prefix));
+        let UnParsedActivitydatawithThreat = union ParsedActivitydata, parsedotheractivitydata_severity
+            | where isnotempty(threatId_s)
+            | lookup Threatsdata on $left.threatId_s == $right.threatInfo_threatId_s
```

Comment:
@jayeshprajapaticrest
can you help me understand why there's a need to do lookup? I saw the raw logs you attached, threatInfo_threatId_s is always empty. Also, Logs where 'threatId_s' is present we have other fields too.

Reply:
As discussed in today's call, there are 2 different records for activity and threat which we have combined using threatId column( threatId_s available in "Activities." data with threatInfo_threatId_s available in "Threats." data). 
As both records are available in a single table(SentinelOne_CL), All threat-related information is available in the Threat record only while all audit activity-related information is available in the Activity record.

In sample data, we have already added Threat-related records at the end in ingested logs and also added in raw logs.